be/src/fs/fs_s3.cpp

```diff
@@ -106,11 +108,14 @@ class S3ClientFactory {
 
     class ClientCacheKey {
     public:
-        ClientConfiguration config;
-        AWSCloudConfiguration aws_cloud_configuration;
+        ClientConfigurationPtr config;
+        AWSCloudConfigurationPtr aws_cloud_configuration;
 
         bool operator==(const ClientCacheKey& rhs) const {
-            return config == rhs.config && aws_cloud_configuration == rhs.aws_cloud_configuration;
+            if (config && rhs.config && aws_cloud_configuration && rhs.aws_cloud_configuration) {
+                return *config == *(rhs.config) && *aws_cloud_configuration == *(rhs.aws_cloud_configuration);
+            }
+            return !config && !rhs.config && !aws_cloud_configuration && !rhs.aws_cloud_configuration;
```

Comment:
They equal when `!config && !rhs.config && aws_cloud_configuration && rhs.aws_cloud_configuration && *aws_cloud_configuration == *(rhs.aws_cloud_configuration)` is true.
So you should change the logic and compare `config` and `aws_cloud_configuration` separately.